### ISSUE TITLE
`logit_scale` and resume

### ISSUE THREAD

**Issue by mitchellnw** [mlfoundations/open_clip#305]
I'm noticing that `logit_scale` will steeply change directions on resume. Probably not a huge deal as it stays fairly close to 100, but worth tracking this issue in case others encounter and its indicative of some other problem with how we clip?

<img width="673" alt="Screen Shot 2022-12-19 at 2 44 33 PM" src="https://user-images.githubusercontent.com/41302489/208541328-342403c6-e5e0-435d-acd8-e8e3876dd7d0.png">

**Comment by rwightman** [comment]
@mitchellnw that's interesting, I haven't observed that before in previous runs. I went back to check across some old resumes and even in the overlap (where there were logs before a mid-epoch crash), the resumed part that overlaps was pretty much the same +/ .2-.3

I wonder if there is something re G. I have seen sudden drops though from 100 to high 80s without any sort of resume, and recovery.

**Comment by mitchellnw** [comment]
closing because the hypothesis is that it relates to a filesystem issue which should not affect most

**Comment by mitchellnw** [comment]
Alright I think I figured out what's happening here, and it does not relate to a filesystem issue! Just in case anyone wants a puzzle to think about, I'll share one more piece of evidence which is the graphs for `logit_scale` and `samples/s`. Will then post what I think the answer is later :)

<img width="400" alt="Screen Shot 2022-12-25 at 3 41 25 PM" src="https://user-images.githubusercontent.com/41302489/209484810-ee7a2e61-63e3-4265-9b7e-af7fe3a56007.png">

**Comment by rwightman** [comment]
@mitchellnw coming back to this one, I don't feel the explanation makes sense, logit scale dips should have no correlation with the end of SCI in terms of dataset randomness.

Each dataloader worker process across each train process (1 per GPU) is sampling with replacement a shard to read from and then reading the samples from that shard and shuffling them within a smaller shuffle buffer. There should be no noteworthy difference in data distribution that relates to the position in the checkpoint interval, even if the samples across the shards weren't shuffled (I believe @rom1504 said they were at some point?).
